presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -248,7 +248,8 @@ private RelationPlan createInsertPlan(Analysis analysis, Insert insertStatement)
             Symbol output = symbolAllocator.newSymbol(column.getName(), column.getType());
             int index = insert.getColumns().indexOf(columns.get(column.getName()));
             if (index < 0) {
-                assignments.put(output, new NullLiteral());
+                Expression cast = new Cast(new NullLiteral(), column.getType().getTypeSignature().toString());
```

Comment:
I checked the spec. In this case, the correct value to insert is the default value associated with the column. Presto doesn't support default values in general, so this is probably ok to do for now. It could result in unexpected data in certain connectors, though (e.g., any jdbc-based connector with tables that contain such columns)